lib/cloud/clients.go

```diff
@@ -231,143 +234,133 @@ type azureClients struct {
 	azureRunCommandClients azure.ClientMap[azure.RunCommandClient]
 }
 
-// GetAWSSession returns AWS session for the specified region.
-func (c *cloudClients) GetAWSSession(region string) (*awssession.Session, error) {
-	c.mtx.RLock()
-	if session, ok := c.awsSessions[region]; ok {
-		c.mtx.RUnlock()
-		return session, nil
-	}
-	c.mtx.RUnlock()
-	return c.initAWSSession(region)
+type SessionOptions struct {
+	Region           string
+	Session          *awssession.Session
+	RoleARN          string
+	ExternalID       string
+	CacheAssumedRole bool
```

Comment:
nit missing docs.

Reply:
renamed this and added docs in fc800e15f75f7ed8f38b88b6aba00a756c66d144